### ISSUE TITLE
Sphere Light

### ISSUE THREAD

**Issue by Doc-Sun** [stanavdb/cosmere-rpg#113]
**Is your feature request related to a problem? Please describe.**  
No real problem, more thematic and fitting to the setting. We should have infused spheres as a light source option

**Describe the solution you'd like**  
Have some way to have a simple "Infused Spheres Light" or make spheres have a toggle to say show light in a bright/dim radius and make it colored based on the color of sphere used even. It is used so often in the books it really should be there.

**Describe alternatives you've considered**  
user could always install a mod(such as torch) but id prefer to keep as much utility into the base game as possible 

**Additional context**  
Allowing for something to this effect 

![image](https://github.com/user-attachments/assets/e984822a-3520-4f3f-91ff-0fe5e878463c)
****

**Checklist:**
- [ ] I have searched the existing issues to ensure this feature has not already been requested.

**Comment by MangoFVTT** [comment]
Note that sphere light is a Stormlight specific thing, not a Cosmere generic thing, so putting this in the base system would not make much sense unless we simply want to have generic light presets that users can select (which is what I believe Torch does)

**Comment by Doc-Sun** [comment]
Yep. Like I said no real problem (and clearly I can do it with torch) but having item context based switches in general that show up on a sheet when in inventory or equipped would be something good to have as code and definitely as a base item. If any light source from any cosmere universe such as a sphere or a lantern etc, had a pre coded toggle would be a nice feature. If the majority of the coding team think it's not worth it that's cool. Just think it might be a nice have especially for expandability in the future.

**Comment by MangoFVTT** [comment]
The most straightforward way to do it without a huge workload would be to open up token light attributes to effects, so that an Item can add a "light source" effect that modifies those dynamically when equipped. This module does that for dnd5e as a point of reference: https://github.com/kandashi/Active-Token-Lighting

**Comment by Doc-Sun** [comment]
Yep that's kinda exactly what I'm looking for an inactive effect that can be turned on. (I was looking but didn't see lighting as an option so if it can be added that solves that!) Can be put on all the different possible sources of light!

**Comment by Dez384** [comment]
The easiest way to implement this is with a macro. Here's a quick hack of the community Token Light Picker macro:
`function tokenUpdate(data) {
  canvas.tokens.controlled.map(token => token.document.update({light: data}));
}

let torchAnimation = {"type": "torch", "speed": 1, "intensity": 1, "reverse": false};

let dialogEditor = new Dialog({
  title: `Sphere Lantern Picker`,
  content: `How full is your sphere lantern?`,
  buttons: {
    empty: {
      label: `Empty`,
      callback: () => {
        tokenUpdate({"dim": 0, "bright": 0, "angle": 360, "color": "#000000", "luminosity": 0.5});
        dialogEditor.render(true);
      }
    },
    onesphere: {
      label: `Single Sphere`,
      callback: () => {
        tokenUpdate({"dim": 5, "bright": 0, "angle": 360, "color": "#e3e3e3", "luminosity": 0.25, "animation": {type: "torch"}});
        dialogEditor.render(true);
      }
    },
    quarter: {
      label: `Quarter-filled`,
      callback: () => {
        tokenUpdate({"dim": 15, "bright": 5, "angle": 360, "color": "#e3e3e3", "luminosity": 0.25, "animation": {type: "torch"}});
        dialogEditor.render(true);
      }
    },
    half: {
      label: `Half-full`,
      callback: () => {
        tokenUpdate({"dim": 30, "bright": 15, "angle": 360, "color": "#e3e3e3", "luminosity": 0.25, "animation": {type: "torch"}});
        dialogEditor.render(true);
      }
    },	
    full: {
      label: `Filled`,
      callback: () => {
        tokenUpdate({"dim": 60, "bright": 30, "angle": 360, "color": "#e3e3e3", "luminosity": 0.25, "animation": {type: "torch"}});
        dialogEditor.render(true);
      }
    },
    close: {
      icon: "<i class='fas fa-x'></i>",
      label: `Close`
    },
  },
  default: "close",
  close: () => {}
});

dialogEditor.render(true)`

Changing the buttons to dropdowns and adding colors would be logical next steps.

**Comment by Doc-Sun** [comment]
My friend. That is beautiful! I could totally see many uses for that!

**Comment by Dez384** [comment]
I've made a macro quick fix for this issue; it can be found in the [Cosmere Useful Macros](https://github.com/Dez384/Cosmere_Useful_Macros) module.

Here is the text for the script:
[Sphere Lantern Picker.txt](https://github.com/user-attachments/files/17613345/Sphere.Lantern.Picker.txt)
